BL70x BLE does not build with M16S1 or M0S1S variant

### Describe the bug

Missing bl_rand implemention, this seem to be a regression

### Regression

- [ ] This is a regression.

### Steps to reproduce

- build BLE thing with `-DCONFIG_BFLB_BL70X_BLE_M16S1=y` or `-DCONFIG_BFLB_BL70X_BLE_M0S1S=y`
- build failure

### Relevant log output

```shell
[2/322] Generating include/generated/zephyr/version.h
-- Zephyr version: 4.4.99 (/home/vdragon/dev/zephyr_upstreaming/zephyr), build: v4.4.0-3337-g07e34e68d340
[317/322] Linking C executable zephyr/zephyr_pre0.elf; Logical command for additional byproducts on target: zephyr_pre0
FAILED: [code=1] zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/vdragon/dev/zephyr_upstreaming/zephyr/build/zephyr/zephyr_pre0.map 
: && ccache /opt/zephyr-sdk-1.0.1/gnu/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-gcc  -gdwarf-4 -Os zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/riscv/core/offsets/offsets.c.obj  -T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map,/home/vdragon/dev/zephyr_upstreaming/zephyr/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/riscv/core/libarch__riscv__core.a  zephyr/lib/libc/picolibc/liblib__libc__picolibc.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/lib/net_buf/liblib__net_buf.a  zephyr/subsys/portability/posix/c_lib_ext/libsubsys__portability__posix__c_lib_ext.a  zephyr/subsys/random/libsubsys__random.a  zephyr/subsys/usb/device_next/libsubsys__usb__device_next.a  zephyr/subsys/bluetooth/common/libsubsys__bluetooth__common.a  zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a  zephyr/drivers/usb/udc/libdrivers__usb__udc.a  zephyr/drivers/usb/common/buf/libdrivers__usb__common__buf.a  zephyr/drivers/bluetooth/libdrivers__bluetooth.a  zephyr/drivers/cache/libdrivers__cache.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/entropy/libdrivers__entropy.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/hwinfo/libdrivers__hwinfo.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  zephyr/drivers/regulator/libdrivers__regulator.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/syscon/libdrivers__syscon.a  zephyr/drivers/timer/libdrivers__timer.a  modules/mbedtls/mbedtls/library/libmbedtls.a  modules/mbedtls/libmodules__mbedtls.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L/home/vdragon/dev/zephyr_upstreaming/zephyr/build/zephyr  zephyr/libcode_relocation_source_lib.a  zephyr/arch/common/libisr_tables.a  modules/mbedtls/mbedtls/library/libmbedx509.a  modules/mbedtls/mbedtls/tf-psa-crypto/core/libtfpsacrypto.a  -fuse-ld=bfd  -mabi=ilp32f  -march=rv32imafc_zicsr_zifencei  -mcmodel=medlow  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -znoexecstack  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie  -Wl,--undefined=_sw_isr_table  -Wl,--undefined=_irq_vector_table  -specs=picolibc.specs  -DPICOLIBC_DOUBLE_PRINTF_SCANF  -Wl,--no-relax  -T/home/vdragon/dev/zephyr_upstreaming/zephyr/soc/bflb/bl70x/bflb_rom_symbols.ld  /home/vdragon/dev/zephyr_upstreaming/modules/hal/bouffalolab/zephyr/blobs/lib/bl70x/libblecontroller_bl702_m16s1.a  /home/vdragon/dev/zephyr_upstreaming/modules/hal/bouffalolab/zephyr/blobs/lib/bl70x/libbl702_rf.a -L"/opt/zephyr-sdk-1.0.1/gnu/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/14.3.0/rv32imafc_zicsr_zifencei/ilp32f/space" -lc -lgcc && cd /home/vdragon/dev/zephyr_upstreaming/zephyr/build/zephyr && /usr/bin/cmake -E true
/opt/zephyr-sdk-1.0.1/gnu/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/14.3.0/../../../../riscv64-zephyr-elf/bin/ld.bfd: /home/vdragon/dev/zephyr_upstreaming/modules/hal/bouffalolab/zephyr/blobs/lib/bl70x/libblecontroller_bl702_m16s1.a(hci_util.o): in function `ble_rand':
/home/weiyin/workspace/gitlab/bl_iot_sdk/components/network/ble/blecontroller/modules/common/api/co_math.h:166:(.text.ble_rand+0x0): undefined reference to `bl_rand'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- linux
- mainline latest
- sdk 1.0.1

### Additional Context

_No response_

--- a/drivers/bluetooth/hci/Kconfig.bflb
+++ b/drivers/bluetooth/hci/Kconfig.bflb
@@ -106,8 +106,19 @@ config BFLB_BL70X_BLE_M1S1
 
 config BFLB_BL70X_BLE_M16S1
 	bool "m16s1: All roles, 16 master + 1 slave"
+	help
+	  16 Central + 1 Peripheral connections.
+	  Uses 16KB exchange memory (vs 8KB for other variants).
+	  The memory size must be correctly configured in the DTS (sram0_bt_em16k).
 
 endchoice
+
+config BFLB_BL70X_BLE_EM_16K
+	bool
+	default y if BFLB_BL70X_BLE_M16S1
+	help
+	  Use 16KB exchange memory for BLE controller.
+	  Only needed for the m16s1 variant.
 
 endif # BT_BFLB_BL70X
 
@@ -231,6 +242,7 @@ config BFLB_BL70XL_BLE_M8S1P
 	help
 	  8 BLE connections, master or slave. All controller code in flash.
 	  Supports PDS. Uses 16KB exchange memory (vs 8KB for other variants).
+	  The memory size must be correctly configured in the DTS (sram0_bt_em16k).
 
 endchoice
 

--- a/dts/riscv/bflb/bl70x.dtsi
+++ b/dts/riscv/bflb/bl70x.dtsi
@@ -427,10 +427,15 @@
 			#size-cells = <1>;
 			ranges = <0x0 0x42018000 DT_SIZE_K(96)>;
 
-			/* 88 KB — top 8 KB reserved for BLE exchange memory */
+			/* 88 KB — top 8 KB or 16 KB reserved for BLE exchange memory */
 			sram0_bt: sram@0 {
 				compatible = "mmio-sram";
 				reg = <0x0 DT_SIZE_K(88)>;
+			};
+
+			sram0_bt_em16k: sram_16k@0 {
+				compatible = "mmio-sram";
+				reg = <0x0 DT_SIZE_K(80)>;
 			};
 		};
 	};

--- a/soc/bflb/bl70x/bflb_platform_shim.c
+++ b/soc/bflb/bl70x/bflb_platform_shim.c
@@ -458,6 +458,11 @@ void srandom(unsigned int seed)
 	/* TRNG-based, no seeding needed */
 }
 
+int bl_rand(void)
+{
+	return (int)sys_rand32_get();
+}
+
 /* Stack info for blob */
 
 /*

--- a/soc/bflb/bl70x/soc.c
+++ b/soc/bflb/bl70x/soc.c
@@ -32,9 +32,11 @@ void soc_early_init_hook(void)
 	tmp = tmp & HBN_REG_EN_HW_PU_PD_UMSK;
 	sys_write32(tmp, HBN_BASE + HBN_IRQ_MODE_OFFSET);
 
-	/* Configure "seam" (BLE exchange memory): 8KB with BLE, 0KB otherwise */
+	/* Configure "seam" (BLE exchange memory): 8KB or 16KB with BLE, 0KB otherwise */
 	tmp = sys_read32(GLB_BASE + GLB_SEAM_MISC_OFFSET);
-#if defined(CONFIG_BT_BFLB_BL70X)
+#if defined(CONFIG_BFLB_BL70X_BLE_EM_16K)
+	tmp = (tmp & GLB_EM_SEL_UMSK) | (0xFU << GLB_EM_SEL_POS);
+#elif defined(CONFIG_BT_BFLB_BL70X)
 	tmp = (tmp & GLB_EM_SEL_UMSK) | (3U << GLB_EM_SEL_POS);
 #else
 	tmp = (tmp & GLB_EM_SEL_UMSK) | (0U << GLB_EM_SEL_POS);

